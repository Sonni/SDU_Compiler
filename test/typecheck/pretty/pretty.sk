type t_int      = int;
type t_rec      = record of {value: int, next: t_rec};
var rec: t_rec;
var i: int;

func add(a: int, b: int): int
    i++;
    return a + b;
end add

i = 0;
while i < 10 do
{
    rec.value += add(3, 4);
}

if true then
    write 1;
else
    write 0;