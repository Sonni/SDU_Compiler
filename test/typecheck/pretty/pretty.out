func add(a: int, b: int): int
    i = (i:int + 1:int):int;
    return (a:int + b:int):int;
end add

var i: int;
var rec: t_rec:record;
type t_rec = record of {value: int, next: t_rec:record};
type t_int = int;
i = 0:int;
while (i:int < 10:int):bool
{
    rec.value = add(3:int, 4:int):int;
}
if true:bool then
    write 1:int;
else
    write 0:int;